Second spawn warden/detective gets other uniform and also some additional items

DrDiasyl opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

If there can be more then 1 detective or warden (Like if it in map configs or some global config change) second warden/detective will get some items and diffirent set of clothings, because of copy-pasted part of the lawyer job-type code.

- Second detective gets a noir suit with trenchcoat and grey fedora. **Also he gets disabler on spawn, this is to make sure detective have any kind of sidearm to protect himself.**
- Second warden gets a formal jumpsuit, warden's jacket and beret. **Also he gets full secbelt and gripper gloves as krav maga replacment.**

## Why It's Good For The Game

This will be good for: maps with several detectives/wardens, servers with changed job capacity configs. Also giving second warden/detective other uniform makes first and second one more distinct from each other, and decrease possible confusion. And also solution to the detective sidearm problem, now second detective will have a disabler which is not a revolver but at least something.

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. -->

:cl: DrDiasyl (DrTuxedo)
balance: Second detective now spawns in a noir suit and with a disabler.
balance: Second warden now spawns in a warden jacket and beret, and with a full secbelt and gripper gloves.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DrDiasyl committed:
Makes second detective spawn with a Noir uniform

DrDiasyl committed:
Makes second warden spawn with Blue uniform

DrDiasyl committed:
Second warden also gets a full secbelt.

TheBonded commented:
I feel like instead of second revolver or set of krav mega gloves, you should give the second detective a disabler and the second warden a set of gripper/rocket gloves. There's only one of each for a reason

DrDiasyl committed:
Now second warden PDA spawn in the pocket.

Handcuffs now spawn in bag and PDA in pocket for a second warden, otherwise PDA wouldn't assign to the second warden.

DrDiasyl commented:
> I feel like instead of second revolver or set of krav mega gloves, you should give the second detective a disabler and the second warden a set of gripper/rocket gloves. There's only one of each for a reason

Well, you actually have a point. But just those items are like the part of this jobs, their trademark. But i can do that yeah.

TheBonded commented:
> > I feel like instead of second revolver or set of krav mega gloves, you should give the second detective a disabler and the second warden a set of gripper/rocket gloves. There's only one of each for a reason
> 
> Well, you actually have a point. But just those items are like the part of this jobs, their trademark. But i can do that yeah.

Similar to how second captains don't get another sabre, or second CMOs don't get a second implant+hardsuit, or even for non heads like second chaplains not getting extra null rods or holy beacons. There are just some items that are set to have only one of them exist for balance. You should probably ask a maintainer to be sure, they might be fine with having a second set of krav mega and a second revolver

DrDiasyl commented:
> > > I feel like instead of second revolver or set of krav mega gloves, you should give the second detective a disabler and the second warden a set of gripper/rocket gloves. There's only one of each for a reason
> > 
> > 
> > Well, you actually have a point. But just those items are like the part of this jobs, their trademark. But i can do that yeah.
> 
> Similar to how second captains don't get another sabre, or second CMOs don't get a second implant+hardsuit, or even for non heads like second chaplains not getting extra null rods or holy beacons. There are just some items that are set to have only one of them exist for balance. You should probably ask a maintainer to be sure, they might be fine with having a second set of krav mega and a second revolver

The thing is that there can be only head no matter what, unlike wardens and detectives. Oh and also about chaplain, if chaplain already selected a nullrod type and a religion the latejoin chaplain will get a nullrod type of first chaplain.

TheBonded commented:
> > > > I feel like instead of second revolver or set of krav mega gloves, you should give the second detective a disabler and the second warden a set of gripper/rocket gloves. There's only one of each for a reason
> > > 
> > > 
> > > Well, you actually have a point. But just those items are like the part of this jobs, their trademark. But i can do that yeah.
> > 
> > Similar to how second captains don't get another sabre, or second CMOs don't get a second implant+hardsuit, or even for non heads like second chaplains not getting extra null rods or holy beacons. There are just some items that are set to have only one of them exist for balance. You should probably ask a maintainer to be sure, they might be fine with having a second set of krav mega and a second revolver
> 
> The thing is that there can be only head no matter what, unlike wardens and detectives. Oh and also about chaplain, if chaplain already selected a nullrod type and a religion the latejoin chaplain will get a nullrod type of first chaplain.

About the chaplain, that's actually kinda cool, though the same doesn't apply for their heroic beacon armour. I'm pretty sure clowns only get one stamp in their toy box though (or am I thinking of something else?). But yeah, I just feel like krav mega gloves and the revolver are set to only have one, and spawn in the locker instead of spawning on their person, for balance reasons.

DrDiasyl commented:
> > > > > I feel like instead of second revolver or set of krav mega gloves, you should give the second detective a disabler and the second warden a set of gripper/rocket gloves. There's only one of each for a reason
> > > > 
> > > > 
> > > > Well, you actually have a point. But just those items are like the part of this jobs, their trademark. But i can do that yeah.
> > > 
> > > 
> > > Similar to how second captains don't get another sabre, or second CMOs don't get a second implant+hardsuit, or even for non heads like second chaplains not getting extra null rods or holy beacons. There are just some items that are set to have only one of them exist for balance. You should probably ask a maintainer to be sure, they might be fine with having a second set of krav mega and a second revolver
> > 
> > 
> > The thing is that there can be only head no matter what, unlike wardens and detectives. Oh and also about chaplain, if chaplain already selected a nullrod type and a religion the latejoin chaplain will get a nullrod type of first chaplain.
> 
> About the chaplain, that's actually kinda cool, though the same doesn't apply for their heroic beacon armour. I'm pretty sure clowns only get one stamp in their toy box though (or am I thinking of something else?). But yeah, I just feel like krav mega gloves and the revolver are set to only have one, and spawn in the locker instead of spawning on their person, for balance reasons.

Well every clown gets a stamp. But krav magas are kinda too cool to be only one thing on station. But revolver.. it's like not that good and bad, i think you can make two of them.

DrDiasyl committed:
Second detective now gets disabler not revolver.

Balance issues

DrDiasyl committed:
Second warden now gets gripper gloves not the krav

Balance issue

DrDiasyl commented:
@TheBonded well i have heard you, and after talking with fellow spessmans i got that having second or more krav-maga/revolvers is a bad idea. Because on some servers like Fulp it can cause opening of like 20 detective slots getting 20 revolvers.

TheBonded commented:
> @TheBonded well i have heard you

Only good things, I hope

>and after talking with fellow spessmans i got that having second or more krav-maga/revolvers is a bad idea. Because on some servers like Fulp it can cause opening of like 20 detective slots getting 20 revolvers.

That was the worry, with a disabler and gripper gloves it makes sure they don't have to take anything from the armoury/spend their credits to get something they should start with, but also doesn't make it possible to get a bunch of revolvers or krav-maga gloves.

DrDiasyl commented:
OOPS, THIS WAS PURE ACCIDENT

optimumtact commented:
the detective has a revolver because it's thematic, giving the second one a disabler is like replacing ian with a cardboard cutout of ian and saying it's the same thing.

LemonInTheDark commented:
Adding new stuff like this always comes with some level of maintenance cost. None of our maps currently run double detective or warden, so why is it a worthwhile change for us to maintain?

optimumtact commented:
wait we don't even have double detective?

OP did you get lost on the way to fulpstations codebase?

TheBonded commented:
> wait we don't even have double detective?
> 
> OP did you get lost on the way to fulpstations codebase?

Currently two detective slots are testmerged on manuel

DrDiasyl commented:
> wait we don't even have double detective?
> 
> OP did you get lost on the way to fulpstations codebase?

#61400 well this PR exists. And also there is some downstreams with second detectives, and a possible chance of some maps being created with two detectives in configs.

DrDiasyl commented:
> Adding new stuff like this always comes with some level of maintenance cost. None of our maps currently run double detective or warden, so why is it a worthwhile change for us to maintain? Outside of if the change is a good idea or not

I don't really see what is here will be maintained so hardly

DrDiasyl commented:
> the detective has a revolver because it's thematic, giving the second one a disabler is like replacing ian with a cardboard cutout of ian and saying it's the same thing.

Fair enough, but just some people got scared about opening somehow detective slots and getting thousands of revolvers

DrDiasyl commented:
> > wait we don't even have double detective?
> > 
> > OP did you get lost on the way to fulpstations codebase?
> 
> Currently two detective slots are testmerged on manuel

Oh epic

DrDiasyl commented:
Oh and by the way, after reading #61400 i thought about making second detective distinctly a more of a partner. Like the way second cook doesn't have a chef hat and apron, but a just apron and a white soft hat.

JohnFulpWillard commented:
I have a feel points I wanted to make
1. Warden/Detectives aren’t command, command can’t have new jobs of them opened outside of admins or bureaucracy error.
2. Detectives can still get new guns via the mail, this just makes additional detectives not have to rely on their mail having one for them.
3. Since this was discussed earlier, all chaplains have a null rod, but the chaplain armor is from the vending machine, and has riot gear level armor, which is already pretty good for some random service job.

DrDiasyl commented:
> I have a feel points I wanted to make
> 1. Warden/Detectives aren’t command, command can’t have new jobs of them opened outside of admins or bureaucracy error.
> 2. Detectives can still get new guns via the mail, this just makes additional detectives not have to rely on their mail having one for them.
> 3. Since this was discussed earlier, all chaplains have a null rod, but the chaplain armor is from the vending machine, and has riot gear level armor, which is already pretty good for some random service job.

Ah i see. But what you think of that idea of making second detective more of a junior a helper?

GuillaumePrata commented:
Almost sure you will need to add `pda_slot = ITEM_SLOT_LPOCKET` somewhere there to not break the auto linking.

I also think you should split this in 2 PRs, most people will agree with the different uniforms being something neat but extra gear is odd.
Security already has an issue of spawning with a lack of equipment and being an easy loot pinata before they get to the brig, late join sec sucks because of this, you don't have glasses and only a baton, but not if you are the second warden or detective that get buffs that all other officers lack.

DrDiasyl commented:
> Almost sure you will need to add `pda_slot = ITEM_SLOT_LPOCKET` somewhere there to not break the auto linking.
> 
> I also think you should split this in 2 PRs, most people will agree with the different uniforms being something neat but extra gear is odd.
> Security already has an issue of spawning with a lack of equipment and being an easy loot pinata before they get to the brig, late join sec sucks because of this, you don't have glasses and only a baton, but not if you are the second warden or detective that get buffs that all other officers lack.
Well i will add pda_slot thing. And also about extra gear, just the thing is warden and detective have only one locker. When sec officer **(Which also gets a disabler roundstart)** can just get to brig and take stuff from locker, second detective/warden can't do that.

LemonInTheDark commented:
> 
> 
> > Adding new stuff like this always comes with some level of maintenance cost. None of our maps currently run double detective or warden, so why is it a worthwhile change for us to maintain? Outside of if the change is a good idea or not
> 
> I don't really see what is here will be maintained so hardly

It's more code that would need to be changed if the structure of jobs is ever modified. There's a cost there.
The lawyer feature you're basing this off was added 10 years ago, and has gone through like 2 different job code refactors.
Used to look like this

https://github.com/tgstation/tgstation/blob/e95d16da16d58c48dcbb036c0e205afd06f732f0/code/game/jobs/jobprocs.dm#L510-L524

I don't want to add things that don't have a good reason to exist. That's not to say that changes like this are bad, but they should exist for a reason that matters to our codebase, not just a downstream.

You need to provide some value to what we have, otherwise it's just code we don't use that we need to upkeep over long stretches of time. I'm unwilling to make the work of the people who come after me harder for no significant reason to them.

DrDiasyl commented:
> > 
> > 
> > > Adding new stuff like this always comes with some level of maintenance cost. None of our maps currently run double detective or warden, so why is it a worthwhile change for us to maintain? Outside of if the change is a good idea or not
> > 
> > I don't really see what is here will be maintained so hardly
> 
> It's more code that would need to be changed if the structure of jobs is ever modified. There's a cost there.
> The lawyer feature you're basing this off was added 10 years ago, and has gone through like 2 different job code refactors.
> Used to look like this
> 
> https://github.com/tgstation/tgstation/blob/e95d16da16d58c48dcbb036c0e205afd06f732f0/code/game/jobs/jobprocs.dm#L510-L524
> 
> I don't want to add things that don't have a good reason to exist. That's not to say that changes like this are bad, but they should exist for a reason that matters to our codebase, not just a downstream.
> 
> You need to provide some value to what we have, otherwise it's just code we don't use that we need to upkeep over long stretches of time. I'm unwilling to make the work of the people who come after me harder for no significant reason to them.

Oh, fair.

DrDiasyl commented:
Oh welp, i just learned about GBP so i probably close this because of how nishe this PR is and how big in negative i will be because of it having a Mechanic tag.
